src/libraries/System.Private.CoreLib/src/System/Half.cs

```diff
@@ -0,0 +1,707 @@
+                {
+                    return CreateHalfNaN(sign, sig << 12); // Shift the significand bits to the left end
+                }
+                return sign ? NegativeInfinity : PositiveInfinity;
+            }
+
+            uint sigHalf = (uint)ShiftRightJam(sig, 38);
+            if ((exp | (int)sigHalf) == 0)
+            {
+                return new Half(sign, 0, 0);
+            }
+            return new Half(RoundPackToHalf(sign, (short)(exp - 0x3F1), (ushort)(sigHalf | 0x4000)));
+        }
+
+        // -----------------------Start of from-half conversions-------------------------
+        public static implicit operator float(Half value)
+        {
+            bool sign = IsNegative(value);
+            int exp = value.Exponent;
+            uint sig = value.Significand;
+
+            if (exp == MaxExponent)
+            {
+                if (sig != 0)
+                {
+                    return CreateSingleNaN(sign, (ulong)sig << 54);
+                }
+                return sign ? float.NegativeInfinity : float.PositiveInfinity;
+            }
+
+            if (exp == 0)
+            {
+                if (sig == 0)
+                {
+                    return CreateSingle(sign ? float.SignMask : 0); // Positive / Negative zero
+                }
+                (exp, sig) = NormSubnormalF16Sig(sig);
+                exp -= 1;
+            }
+
+            return CreateSingle(sign, (byte)(exp + 0x70), sig << 13);
+        }
+
+        public static implicit operator double(Half value)
+        {
+            bool sign = IsNegative(value);
+            int exp = value.Exponent;
+            uint sig = value.Significand;
+
+            if (exp == MaxExponent)
+            {
+                if (sig != 0)
+                {
+                    return CreateDoubleNaN(sign, (ulong)sig << 54);
+                }
+                return sign ? double.NegativeInfinity : double.PositiveInfinity;
+            }
+
+            if (exp == 0)
+            {
+                if (sig == 0)
+                {
+                    return CreateDouble(sign ? double.SignMask : 0); // Positive / Negative zero
+                }
+                (exp, sig) = NormSubnormalF16Sig(sig);
+                exp -= 1;
+            }
+
+            return CreateDouble(sign, (ushort)(exp + 0x3F0), (ulong)sig << 42);
+        }
+
+        // IEEE 754 specifies NaNs to be propagated
+        internal static Half Negate(Half value)
+        {
+            return IsNaN(value) ? value : new Half((ushort)(value.m_value ^ SignMask));
+        }
+
+        private static (int Exp, uint Sig) NormSubnormalF16Sig(uint sig)
+        {
+            int shiftDist = BitOperations.LeadingZeroCount(sig) - 16 - 5;
+            return (1 - shiftDist, sig << shiftDist);
+        }
+
+        #region Utilities
+
+        private static uint ToUInt32(float value)
+                   => (uint)BitConverter.SingleToInt32Bits(value);
```

Comment:
Nit: indenting is inconsistent, e.g. the CreateHalf method below indents four spaces from the start of the signature